### ISSUE TITLE
How to Train robertaclassifier by freezing roberta weights

### ISSUE THREAD

**Issue by suryapa1** [ThilinaRajapakse/simpletransformers#797]
**Describe the bug**
Would like to Train robertaclassifier by freezing Roberta weights and plan to finetune classification layer alone, any code snippet will be really appreciated.

Thanks

**Comment by ThilinaRajapakse** [comment]
[This](https://simpletransformers.ai/docs/tips-and-tricks/#train-custom-parameters-only) example from the docs is for a BERT model but the approach is pretty much the same for RoBERTa.

**Comment by roboserg** [comment]
> [This](https://simpletransformers.ai/docs/tips-and-tricks/#train-custom-parameters-only) example from the docs is for a BERT model but the approach is pretty much the same for RoBERTa.

In your example the code used is:
```
model_args.custom_parameter_groups = [
    {
        "params": ["classifier.weight"],
```

How do I:
1. see a list of names of layers in the model
2. delete and add my own final layer using simpletransformers

I want to train roberta for 3x label NER problem but roberta outputting 9 classes

**Comment by ThilinaRajapakse** [comment]
1. [Docs](https://simpletransformers.ai/docs/tips-and-tricks/#custom-parameter-groups).

>Hint: All Simple Transformers models have a get_named_parameters() method that returns a list of all parameter names in the model.

2. You don't need to delete and add your own layers if you just want to use a NER model with 3 labels. [You can specify your own labels](https://simpletransformers.ai/docs/ner-specifics/#custom-labels).

If you really do need to manually delete/add layers, that's a Pytorch level task.

**Comment by roboserg** [comment]
> 2. You don't need to delete and add your own layers if you just want to use a NER model with 3 labels. [You can specify your own labels](https://simpletransformers.ai/docs/ner-specifics/#custom-labels).

Thanks but I am still confused. I want to do NER with three custom labels, I am already specifying my own labels as you suggest:

```
labels = df_train["labels"].unique().tolist()
print(labels)
-> ['O', 'B-KEY', 'I-KEY']
```
When using [BERT](https://huggingface.co/deepset/gbert-large) it works, but when [using xlm-roberta-base](https://huggingface.co/xlm-roberta-base) or xlm-roberta-large-finetuned-conll03-german I get an error, since, as you said, the number of output parameters will be different. 
1. I just don't understand why does it work for BERT, it has 9 outputs, not 3. 
2. What steps would I need to do to train xlm-roberta-base with 3 outputs for a NER task. You said I won't have to use Pytorch for that. I am confused why it works for BERT but not for Roberta. In both cases the pretrained model has a different number of outputs then my 3 labels, but it works with BERT.

**Comment by ThilinaRajapakse** [comment]
The issue is likely that the `model_type` should be `xlmroberta` and not `roberta` when you are using `xlm-roberta-X` models. However, if the model you are loading is fine-tuned, then you can't change the number of labels after that. A fine-tuned model has a final layer with trained weights which can't be changed with Simple Transformers.

It works with BERT because a model like `bert-base-cased` is _pre-trained_ but not _fine-tuned_ on a specific task. (Also, because you likely set the `model_type` correctly to `bert).
